### ISSUE TITLE
Types not available inside `onUpdate`

### ISSUE THREAD

**Issue by sidharthv96** [ciscoheat/sveltekit-superforms#176]
**Description**
Form's field types aren't available inside onUpdate.

<img width="818" alt="image" src="https://github.com/ciscoheat/sveltekit-superforms/assets/10703445/2e361a40-b973-4637-8896-c59903975666">

<img width="596" alt="image" src="https://github.com/ciscoheat/sveltekit-superforms/assets/10703445/b7da76f7-b570-49dc-9d20-abc414dc509a">

**If applicable, a MRE**

https://stackblitz.com/edit/sveltekit-superforms-bug-reporting-vnk9a7?file=src%2Froutes%2F%2Bpage.svelte

**Comment by ciscoheat** [comment]
The MRE seems incomplete, I get a 500 error when opening it, but I can reproduce it locally. It will be fixed in 1.0, released soon.

**Comment by ciscoheat** [comment]
This occurs if you are sending null, undefined or a data object to `superForm`, without using `superValidate`. In that case you must add a type parameter with the schema type to `superForm`:

```ts
const { form, errors, enhance } = superForm<YourSchema>(...)
```

Please reopen this issue if you have used `superValidate` and still get this problem.

**Comment by sidharthv96** [comment]
@ciscoheat I think reopen only available for contributors. I only have the comment button.

**Comment by ciscoheat** [comment]
In what file are you doing this? `superValidate` is async, so you need to `await superValidate`, but that cannot be made in a `+page.svelte` file, where top-level await is disallowed. Since your form is for a SPA, you need to use a `+page.ts` to load the initial data. See the docs here: https://superforms.vercel.app/concepts/spa#using-pagets-instead-of-pageserverts

In 1.0 you'll have access to `superValidateSync`, so you can put everything in a `+page.svelte` file.

**Comment by sidharthv96** [comment]
Yup, that was the issue. It wasn't clear in the docs why the +page.ts was needed (top level await), and the `superForm(superValidate(loginSchema), ...` call didn't trigger any TS errors, so I totally missed the fact that superValidate returns a promise and jammed it into the svelte file. 

Thank you. 

Do you think `superForm(superValidate(loginSchema), ...` should trigger a TS error? Or is that a valid scenario?
Please close the issue if it's valid.

**Comment by ciscoheat** [comment]
`await superValidate(...)` is a valid scenario, but sending arbitrary data to `superForm` will not be allowed in 1.0, and will trigger a TS error.
